api/wasm/cpp/proxy_wasm_api.h

```diff
@@ -513,6 +513,78 @@ inline bool getMessageValue(std::initializer_list<StringView> parts, T* value_pt
   return value_ptr->ParseFromArray(buf.value()->data(), buf.value()->size());
 }
 
+// Create an expression using a foreign function call.
+inline WasmResult exprCreate(StringView expr, uint32_t* token) {
+  std::string function = "expr_create";
+  char* out = nullptr;
+  size_t out_size = 0;
+  auto result = proxy_call_foreign_function(function.data(), function.size(), expr.data(),
+                                            expr.size(), &out, &out_size);
+  if (result != WasmResult::Ok) {
+    return result;
+  }
+  if (out_size != sizeof(uint32_t)) {
+    return WasmResult::ResultMismatch;
+  }
+  *token = *reinterpret_cast<uint32_t*>(out);
+  return result;
+}
+
+// Evaluate an expression using an expression token.
+inline Optional<WasmDataPtr> exprEval(uint32_t token) {
```

Comment:
Please move these to an separate header file.  Anything built on top of a foreign function should be in its own file, perhaps in a separate directory e.g. contrib.